lib/model-builder.js

```diff
@@ -257,6 +257,12 @@ ModelBuilder.prototype.define = function defineClass(className, properties, sett
       deprecated('Property names containing a dot are not supported. ' +
         'Model: ' + className + ', property: ' + p);
     }
+
+    // Throw error if property name is 'constructor'
+    if (p === "constructor") {
+      throw new Error('Property name should not be constructor in Model: ' +
+       className);
```

Comment:
Thank you Simon, this code is not ready for review LOL,  lots of things haven't been checked. But thanks for reminding me this, I will fix it.